component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/GenericInputValidator.java

```diff
@@ -0,0 +1,151 @@
+package org.wso2.am.analytics.publisher.reporter;
+
+import java.util.AbstractMap;
```

Comment:
Why do we need this class? Validation is a part of each implementation. What purpose does a generic validator serve

Reply:
This is an alternative to the DefaultInputValidator
target event maps structures inside the DefaultInputValidator dont have the userName attribute (which can be used in choreo instances). 
Whereas in the GenericInputValidator, I included the userName attribute (which can be used in other instances except for choreo).

Reply:
Added this to make sure we have a corresponding target event map structure for the source event map structure (since with the current impl source event map structure includes all the attributes. )

Reply:
Then this is not the generic validator. This is another type of a validator isn't it. which is used by ELK and Moseif. What are the other implementations which uses the username apart from Moseif

Reply:
I think Tishan we can safely assume almost every other analytics platform uses the userName.
Datadog uses it for sure.
Implementation done in a way so we can easily get the userName using GenericInputValidator .

Reply:
>> Then this is not the generic validator. This is another type of a validator isn't it.
Yes, Tishan.
I put the term generic in a sense, now the DefaultInputValidator is the specific validator (because this doesn't have the userName and almost every other analytics platform requires userName).
So by "Generic" I meant this is the general validator for any other platform.